File: server/src/main/java/com/smartexpense/api/presentation/controller/UserController.java
package com.smartexpense.api.presentation.controller;

import com.smartexpense.api.application.dto.request.UpdateProfileRequestDto;
import com.smartexpense.api.application.dto.response.UserProfileResponseDto;
import com.smartexpense.api.application.port.in.ProfileUseCase;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
@RequiredArgsConstructor
public class UserController {

    private final ProfileUseCase profileUseCase;

    @GetMapping("/me")
    public ResponseEntity<UserProfileResponseDto> getMyProfile(Authentication authentication) {
        String email = authentication.getName();
        return ResponseEntity.ok(profileUseCase.getProfile(email));
    }

    @PutMapping("/me")
    public ResponseEntity<UserProfileResponseDto> updateProfile(
            Authentication authentication,
            @RequestBody UpdateProfileRequestDto request) {
        String email = authentication.getName();
        return ResponseEntity.ok(profileUseCase.updateProfile(email, request));
    }
}

File: server/src/main/kotlin/com/smartexpense/api/presentation/controller/UserController.kt
package com.smartexpense.api.presentation.controller

import com.smartexpense.api.application.dto.request.UpdateProfileRequestDto
import com.smartexpense.api.application.dto.response.UserProfileResponseDto
import com.smartexpense.api.application.port.`in`.ProfileUseCase
import org.springframework.http.ResponseEntity
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/users")
class UserController(
    private val profileUseCase: ProfileUseCase
) {

    @GetMapping("/me")
    fun getMyProfile(authentication: Authentication): ResponseEntity<UserProfileResponseDto> {
        val email = authentication.name
        return ResponseEntity.ok(profileUseCase.getProfile(email))
    }

    @PutMapping("/me")
    fun updateProfile(
        authentication: Authentication,
        @RequestBody request: UpdateProfileRequestDto
    ): ResponseEntity<UserProfileResponseDto> {
        val email = authentication.name
        return ResponseEntity.ok(profileUseCase.updateProfile(email, request))
    }
}
